root/static/scripts/edit/utility/linkPhrase.js

```diff
@@ -31,7 +31,7 @@ exports.clean = _.memoize(function (linkType, backward) {
         }
 
         return match;
-    });
+    }).trim().replace(/ {2,}/g, ' ');
```

Comment:
That's actually what https://github.com/y-van-z/musicbrainz-server/blob/a486351bec11911b1738d3c09fa251970a4d06a7/root/static/scripts/common/utility/clean.js does and it's already imported in this file. So you can just wrap the return like `return clean(phrase.replace(...));`.